doc/api/esm.md

```diff
@@ -159,6 +156,15 @@ via the paths defined in [`"exports"`][].
 For details on these package resolution rules that apply to bare specifiers in
 the Node.js module resolution, see the [packages documentation](packages.md).
 
+### Mandatory file extensions
+
+A file extension must be provided when using the `import` keyword to resolve
+relative or absolute specifiers. Directory indexes (e.g. `'./startup/index.js'`)
+must also be fully specified.
+
+This behavior matches how `import` behaves in browser environments, assuming a
+typically configured server.
```

Comment:
Can you cite a source for "typically configured server"? Browsers simply don't care one way or the other about extensions, so this statement doesn't make sense.